modmesh/app/euler1d.py

```diff
@@ -61,73 +66,315 @@ def load_app():
+                (self.temperature, 'c'),
+                (self.int_energy, 'k'),
+                (self.entropy, 'm')
         )):
             # Plot multiple data line with same X axis, it need to plot a
             # data line on main axis first
             if i == 0:
-                data.ana, = self.ax.plot(x.copy(), np.zeros_like(x),
-                                         f'{color}-',
-                                         label=f'{data.name}_ana')
-                data.num, = self.ax.plot(x.copy(), np.zeros_like(x),
-                                         f'{color}x',
-                                         label=f'{data.name}_num')
-                self.ax.set_ylabel(f'{data.name} ({data.unit})')
+                data.bind_axis(ax)
+                data.ana, = ax.plot(x.copy(), np.zeros_like(x),
+                                    f'{color}-',
+                                    label=f'{data.name}_ana')
+                data.num, = ax.plot(x.copy(), np.zeros_like(x),
+                                    f'{color}x',
+                                    label=f'{data.name}_num')
+                ax.set_ylabel(f'{data.name} ({data.unit})')
             else:
-                ax_new = self.ax.twinx()
+                ax_new = ax.twinx()
+                data.bind_axis(ax_new)
                 data.ana, = ax_new.plot(x.copy(), np.zeros_like(x),
                                         f'{color}-',
                                         label=f'{data.name}_ana')
                 data.num, = ax_new.plot(x.copy(), np.zeros_like(x),
                                         f'{color}x',
                                         label=f'{data.name}_num')
-                ax_new.spines.right.set_position(("axes", 1 + (i - 1) * 0.07))
-                ax_new.set_ylim(self.ax.get_ylim())
+                ax_new.spines.right.set_position(("axes", 1 + (i - 1) * 0.1))
                 ax_new.set_ylabel(f'{data.name} ({data.unit})')
 
             lines.append(data.ana)
             lines.append(data.num)
 
-        self.ax.legend(lines, [line.get_label() for line in lines])
-        self.ax.set_xlabel("distance (m)")
+        ax.legend(lines, [line.get_label() for line in lines])
+        ax.set_xlabel("distance (m)")
+        ax.grid()
+        canvas.installEventFilter(self)
+
+        # These parameters are the results obtained by my tuning on GUI
+        fig.subplots_adjust(left=0.053, right=0.638, bottom=0.093, top=0.976,
+                            wspace=0.2, hspace=0.2)
 
-    def update_lines(self, shocktube):
-        self.density.update(adata=shocktube.density_field,
-                            ndata=shocktube.svr.density[::2])
-        self.pressure.update(adata=shocktube.pressure_field,
-                             ndata=shocktube.svr.pressure[::2])
-        self.velocity.update(adata=shocktube.velocity_field,
-                             ndata=shocktube.svr.velocity[::2])
+        frameContainer.addWidget(canvas)
```

Comment:
We are mixing `camelCase` and `snake_case` conventions.  We can live with it now.  After we have more code and experience we should come up with a way to reconcile the conflict in styles.

Reply:
I fixed it in https://github.com/solvcon/modmesh/pull/210/commits/0c24c1081ed6b4c386c7d1e95e315336a8987951